server/src/testFixtures/java/org/elasticsearch/repositories/blobstore/BlobStoreTestUtil.java

```diff
@@ -200,48 +218,42 @@ private static void assertSnapshotUUIDs(BlobStoreRepository repository, Reposito
-                            assertThat(indexContainer.listBlobs(),
-                                       hasKey(String.format(Locale.ROOT, BlobStoreRepository.METADATA_NAME_FORMAT, snapshotId.getUUID())));
-                            final IndexMetadata indexMetadata = PlainActionFuture.get(x -> repository.getSnapshotIndexMetadata(snapshotId, indexId, x));
-                            for (Map.Entry<String, BlobContainer> entry : indexContainer.children().entrySet()) {
-                                // Skip Lucene MockFS extraN directory
-                                if (entry.getKey().startsWith("extra")) {
-                                    continue;
-                                }
-                                final int shardId = Integer.parseInt(entry.getKey());
-                                final int shardCount = indexMetadata.getNumberOfShards();
-                                maxShardCountsExpected.compute(
-                                    indexId, (i, existing) -> existing == null || existing < shardCount ? shardCount : existing);
-                                final BlobContainer shardContainer = entry.getValue();
-                                // TODO: we shouldn't be leaking empty shard directories when a shard (but not all of the index it belongs to)
-                                //       becomes unreferenced. We should fix that and remove this conditional once its fixed.
-                                if (shardContainer.listBlobs().keySet().stream().anyMatch(blob -> blob.startsWith("extra") == false)) {
-                                    final int impliedCount = shardId - 1;
-                                    maxShardCountsSeen.compute(
-                                        indexId, (i, existing) -> existing == null || existing < impliedCount ? impliedCount : existing);
-                                }
-                                if (shardId < shardCount && snapshotInfo.shardFailures().stream().noneMatch(
-                                    shardFailure -> shardFailure.index().equals(index) && shardFailure.shardId() == shardId)) {
-                                    final Map<String, BlobMetadata> shardPathContents = shardContainer.listBlobs();
-                                    assertThat(shardPathContents,
-                                               hasKey(String.format(Locale.ROOT, BlobStoreRepository.SNAPSHOT_NAME_FORMAT, snapshotId.getUUID())));
-                                    assertThat(shardPathContents.keySet().stream()
-                                                   .filter(name -> name.startsWith(BlobStoreRepository.INDEX_FILE_PREFIX)).count(), lessThanOrEqualTo(2L));
-                                }
-                            }
-                        }
-                    },
-                    (e) -> {}
-                )
-            );
+            SnapshotInfo snapshotInfo = repository.getSnapshotInfo(snapshotId).get();
```

Comment:
Blocking call `.get` on a Future. This is only for testing therefore i think it's acceptable.